src/App.tsx

```diff
@@ -1,24 +1,78 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useEffect,
+  useState,
+  useMemo,
+  useContext,
+} from 'react';
+// import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { getTodos } from './api/todos';
+// import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { ErrorMessage } from './types/ErrorMessage';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { ErrorNotification }
+  from './components/ErrorNotification/ErrorNotification';
+import { getFilteredTodos } from './utils/getTodos';
+import { TodoContext } from './context/TodoContext';
+import { USER_ID } from './utils/constants';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const {
+    todos,
+    setTodos,
+    errorMessage,
+    setErrorMessage,
+  } = useContext(TodoContext);
+
+  const [filter, setFilter] = useState(Filter.All);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.LoadingError);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage(ErrorMessage.Default);
+      }, 3000);
+    }
+  }, [errorMessage]);
```

Comment:
https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks

Reply:
> https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks

Check the article and don't forget to remove your timers

Btw, you can keep this logic in related component `ErrorNotification`

Reply:
Not fixed 😢